### PR TITLE
batching indexer operations

### PR DESCRIPTION
**Pull request by mmaxim**

### CODE REVIEW

File: go/chat/search/indexer.go [hunk 1]
```diff
@@ -53,13 +59,14 @@ func NewIndexer(g *globals.Context) *Indexer {
...
-		store:        newStore(g),
 		pageSize:     defaultPageSize,
-		stopCh:       make(chan chan struct{}, 10),
+		stopCh:       make(chan struct{}),
 		suspendCh:    make(chan chan struct{}, 10),
 		resumeWait:   time.Second,
 		cancelSyncCh: make(chan struct{}, 100),
 		pokeSyncCh:   make(chan struct{}, 100),
+		clock:        clockwork.NewRealClock(),
+		flushDelay:   15 * time.Second,
```
Comment (joshblum) [inline go/chat/search/indexer.go]: why not higher?

File: go/chat/search/storage.go [hunk 2]
```diff
@@ -68,18 +69,268 @@ func (a *aliasEntry) remove(token string) bool {
...
+	}
+	res = new(tokenEntry)
+	found, err := b.edb.Get(ctx, key, res)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, nil
+	}
+	return res, nil
```
Comment (joshblum) [inline go/chat/search/storage.go]: populate cache  on read

File: go/chat/search/storage.go [hunk 3]
```diff
@@ -68,18 +69,268 @@ func (a *aliasEntry) remove(token string) bool {
...
+	}
+	res = new(aliasEntry)
+	found, err := b.edb.Get(ctx, key, res)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, nil
+	}
+	return res, nil
```
Comment (joshblum) [inline go/chat/search/storage.go]: populate aliasBatch

File: go/chat/search/storage.go [hunk 4]
```diff
@@ -68,18 +69,268 @@ func (a *aliasEntry) remove(token string) bool {
...
+	batch, ok := b.tokenBatch[convID.String()]
+	if ok {
+		delete(batch.tokens, token)
+	}
+	key, err := tokenKey(ctx, b.uid, convID, token, b.keyFn)
+	if err != nil {
+		b.Debug(ctx, "RemoveTokenEntry: failed to get tokenkey: %s", err)
+		return
+	}
+	if err := b.mdb.Delete(key); err != nil {
```
Comment (joshblum) [inline go/chat/search/storage.go]: shouldn't this be `edb`?

File: go/chat/search/storage.go [hunk 5]
```diff
@@ -68,18 +69,268 @@ func (a *aliasEntry) remove(token string) bool {
...
+func (b *batchingStore) RemoveAliasEntry(ctx context.Context, alias string) {
+	b.Lock()
+	defer b.Unlock()
+	delete(b.aliasBatch, alias)
+	key, err := aliasKey(ctx, alias, b.keyFn)
+	if err != nil {
+		b.Debug(ctx, "RemoveAliasEntry: failed to get key: %s", err)
+		return
+	}
+	if err := b.mdb.Delete(key); err != nil {
```
Comment (joshblum) [inline go/chat/search/storage.go]: `edb`?

File: go/chat/search/storage.go [hunk 6]
```diff
@@ -68,18 +69,268 @@ func (a *aliasEntry) remove(token string) bool {
...
+	key := metadataKey(b.uid, convID)
+	res = new(indexMetadata)
+	found, err := b.mdb.GetIntoMsgpack(res, key)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, nil
+	}
+	return res, nil
```
Comment (joshblum) [inline go/chat/search/storage.go]: populate cache

File: go/chat/search/storage.go (hunk 4 above)
Comment (mmaxim) [reply go/chat/search/storage.go]: metadata not stored encrypted.

Comment (mmaxim) [reply go/chat/search/storage.go]: Oh, it doesn't matter here

File: go/chat/search/storage.go (hunk 3 above)
Comment (mmaxim) [reply go/chat/search/storage.go]: It isn't a cache, it is supposed to mirror what you would have gotten off the disk.

File: go/chat/search/indexer.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/search/indexer.go]: This felt fine to me, any higher and I would like to put in a response to mobile app state,.

**Pull request merged**